### PR TITLE
Removed forced optional type used in `Atomic`

### PR DESCRIPTION
**Pull request by VojtaStavik**
### In the PR:
- `value` in `Atomic` is no longer treated as `Optional` by default
- `Atomic` stress tests random crash (hopefully) fixed 🤞 

---

@buh  @b-onc Please review and test this **extremely** carefully. I touched a lot of different parts of the app and could easily introduce some copy-paste/typo kind of bug 🙏

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 # Upcoming
 
+### ⚠️ Breaking Changes
+- `Atomic.get()` no longer returns an optional type if the wrapped type itself is not optional. `Atomic.init(_:)` now always requires the initial value [#241](https://github.com/GetStream/stream-chat-swift/issues/241).
+
```
Comment (b-onc) [inline CHANGELOG.md:11]: Maybe we should merge this into `release/3.0` since it's breaking and touches a lot of different parts?

Comment (VojtaStavik) [reply CHANGELOG.md:11]: I'd like to merge it to `master`. Let me redo it in a bit more compatible way.

File: Tests/Client/Unit Tests/AtomicTests.swift [hunk 2]
```diff
@@ -164,13 +195,15 @@ extension AtomicTests {
             }
 
             for _ in 0...5 {
-                DispatchQueue.random.async {
+                // We need to capture the value explicitly. Without this the tests randomly crashes after
+                // `atomicValue` is assigned to `nil`.
+                DispatchQueue.random.async { [atomicValue] in
```
Comment (VojtaStavik) [inline Tests/Client/Unit Tests/AtomicTests.swift:200]: 10 minutes to identify the problem + 50 minutes figuring out how to fix it = `[atomicValue]`

**Comment by VojtaStavik** [comment]
@buh @b-onc I changed to PR to be much less destructive. I feel much better with merging this to `master` now.

File: Sources/Client/Utils/Atomic.swift [hunk 3]
```diff
@@ -9,6 +9,9 @@
 import Foundation
 
 /// A mutable thread safe variable.
+///
+/// - Note: Even though the value guarded by `Atomic` is thread-safe, the `Atomic` class itself is not. Mutating the instance
+/// itself from multiple threads can cause a crash.
```
Comment (b-onc) [inline Sources/Client/Utils/Atomic.swift:14]: 👍

**Comment by VojtaStavik** [comment]
@b-onc As far as I understand it, capturing it explicitly creates a strong reference to `atomicValue` and keeps it alive until all reads are safely finished. Without `[atomicValue]` it doesn't create the reference and thus the object can get deinited while the read block is accessing it.

**Pull request merged**